src/actions/ast.js

```diff
@@ -126,3 +135,52 @@ export function setSelection(
     });
   };
 }
+
+let hiddenBreakpointLocation = null;
+
+/**
+ * Action dispatched from stepOver action. Receives the position
+ * and depending on where it is paused, sets up hidden breakpoints
+ * and handles resume, stepOver (as needed)
+ * @param position
+ * @returns {function(ThunkArgs)}
+ */
+export function analyzePauseLocation(position: AstLocation) {
+  return async ({ dispatch, getState, client }: ThunkArgs) => {
+    const source = getSelectedSource(getState()).toJS();
+
+    await dispatch({
+      type: "ANALYZE_PAUSE_LOCATION",
+      [PROMISE]: (async function() {
+        if (hiddenBreakpointLocation) {
```

Comment:
I feel like we should break this up so that we have smaller functions for these different scenarios.